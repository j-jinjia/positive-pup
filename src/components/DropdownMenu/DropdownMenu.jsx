import { Link } from "react-router-dom";
import "./DropdownMenu.scss";
import courseData from "../../assets/mockData/courseData";

const DropdownMenu = () => {
  const getColumn = (courseType) => {
    const colData = courseData.filter(
      (courseData) => courseData.courseTypePlural === courseType
    );
    const col = colData.map((courseData, index) => (
      <a
        href={courseData.link}
        key={courseData.labelsArray.join("") + index}
        className="dropdown-menu__column-option"
      >
        {courseData.courseHeading}
      </a>
    ));
    return col;
  };

  return (
    <div className="dropdown-menu">
      <Link to="/services" className="dropdown-menu__link">
        View All
      </Link>
      <div className="dropdown-menu__column">
        <h3 className="dropdown-menu__column-title">Online Courses</h3>
        {getColumn("Online Courses")}
      </div>
      <div className="dropdown-menu__column">
        <h3 className="dropdown-menu__column-title">Group Classes</h3>
        {getColumn("Group Classes")}
      </div>
      <div className="dropdown-menu__column">
        <h3 className="dropdown-menu__column-title">Other</h3>
        {getColumn("Bespoke Packages")}
      </div>
    </div>
  );
};

export default DropdownMenu;
