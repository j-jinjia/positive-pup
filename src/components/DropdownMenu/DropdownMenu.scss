@use "../../styles/_variables.scss" as *;

.dropdown-menu {
  position: absolute;
  top: 2rem;
  border: 1px solid $colour-septenary;
  border-radius: 0.5rem;
  background-color: $colour-secondary;
  display: flex;
  column-gap: 2rem;
  padding: 2rem;
  text-align: left;

  &:hover {
    cursor: default;
  }

  &__link {
    color: $colour-primary;
    font-weight: 600;
  }

  &__column {
    display: flex;
    flex-direction: column;
    font-weight: 500;
    row-gap: 0.7rem;

    &-title {
      color: $colour-quatenary;
      font-weight: inherit;
      margin-bottom: 0.55rem;
      text-decoration: none;
    }

    &-option {
      text-decoration: none;
      color: $colour-quinary;
    }
  }

  &__button {
    position: absolute;
    background: none;
    border: none;
    top: 0;
    right: 0;
    margin: 0.75rem;
    &-close {
      width: 1rem;
      &:hover {
        cursor: pointer;
      }
    }
  }
}
