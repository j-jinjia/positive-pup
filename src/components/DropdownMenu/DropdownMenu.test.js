import { customRender } from "../../utils/testUtils";
import DropdownMenu from "./DropdownMenu";

it("should match the DropdownMenu component snapshot", () => {
  const { container } = customRender(<DropdownMenu />);

  expect(container).toMatchSnapshot();
});
