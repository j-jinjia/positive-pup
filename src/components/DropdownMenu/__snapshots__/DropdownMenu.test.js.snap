// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the DropdownMenu component snapshot 1`] = `
<div>
  <div
    class="dropdown-menu"
  >
    <a
      class="dropdown-menu__link"
      href="/services"
      target="_parent"
    >
      View All
    </a>
    <div
      class="dropdown-menu__column"
    >
      <a
        class="dropdown-menu__column-title"
        href="/services/Online Courses"
        target="_parent"
      >
        Online Courses
      </a>
      <a
        class="dropdown-menu__column-option"
        href="https://www.thepositivepup.teachable.com/"
        rel="noreferrer"
        target="_blank"
      >
        Chaos to Calm
      </a>
      <a
        class="dropdown-menu__column-option"
        href="https://www.thepositivepup.teachable.com/"
        rel="noreferrer"
        target="_blank"
      >
        Rapid Recall
      </a>
    </div>
    <div
      class="dropdown-menu__column"
    >
      <a
        class="dropdown-menu__column-title"
        href="/services/Group Classes"
        target="_parent"
      >
        Group Classes
      </a>
      <a
        class="dropdown-menu__column-option"
        href="https://www.thepositivepup.teachable.com/"
        rel="noreferrer"
        target="_blank"
      >
        Little Stars Course
      </a>
      <a
        class="dropdown-menu__column-option"
        href="https://www.thepositivepup.teachable.com/"
        rel="noreferrer"
        target="_blank"
      >
        Rising Stars Course
      </a>
      <a
        class="dropdown-menu__column-option"
        href="https://www.thepositivepup.teachable.com/"
        rel="noreferrer"
        target="_blank"
      >
        Little Stars Course
      </a>
    </div>
    <div
      class="dropdown-menu__column"
    >
      <a
        class="dropdown-menu__column-title"
        href="/services/Bespoke Packages"
        target="_parent"
      >
        Other
      </a>
      <a
        class="dropdown-menu__column-option"
        href="https://www.thepositivepup.teachable.com/"
        rel="noreferrer"
        target="_blank"
      >
        1:1 Personalised Training
      </a>
      <a
        class="dropdown-menu__column-option"
        href="https://www.thepositivepup.teachable.com/"
        rel="noreferrer"
        target="_blank"
      >
        Premium Puppy
      </a>
      <a
        class="dropdown-menu__column-option"
        href="https://www.thepositivepup.teachable.com/"
        rel="noreferrer"
        target="_blank"
      >
        Bespoke Packages
      </a>
    </div>
    <button
      class="dropdown-menu__button"
    >
      <img
        alt="close dropdown"
        class="dropdown-menu__button-close"
        src="close-menu.svg"
      />
    </button>
  </div>
</div>
`;
